framework/core/hpp_instance.cpp

```diff
@@ -21,11 +21,23 @@
 #include <core/hpp_physical_device.h>
 #include <volk.h>
 
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+#	define USE_VALIDATION_LAYERS 1
+#endif
+
+#if defined(USE_VALIDATION_LAYERS) && (defined(VKB_VALIDATION_LAYERS_GPU_ASSISTED) || defined(VKB_VALIDATION_LAYERS_BEST_PRACTICES) || defined(VKB_VALIDATION_LAYERS_SYNCHRONIZATION))
+#	define USE_VALIDATION_LAYER_FEATURES 1
+#endif
+
+#ifdef USE_VALIDATION_LAYERS
+#	define USE_VULKAN_LOGGER
+#endif
+
 namespace vkb
 {
 namespace
 {
-#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+#ifdef USE_VULKAN_LOGGER
```

Comment:
Do we ever envision this being enabled independently of USE_VALIDATION_LAYERS?  If not, then maybe let's just use USe_VALIDATION_LAYERS as the test here.  Makes it slightly less complex and doesn't create a possibility of someone accidentally enabling code that might not work independent of other macros.

Reply:
I doubt we ever will. However, having `USE_VULKAN_LOGGER` highlights where code is more specifically logger related and not validation layer related. After all, the logger can be used by other layers to communicate with developers. The logger macro should work independently to `USE_VALIDATION_LAYERS`, right now we just didn't need that functionality in order to fix the bug on main